### PR TITLE
Feature/bug compile with without itkgpu

### PR DESCRIPTION
**Pull request by lcoram**
Should compile if do / do not have gpu turned on in ITK

### CODE REVIEW

File: VesselSegmentation/Logic/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +7,7 @@ set(${KIT}_EXPORT_DIRECTIVE "VTK_SLICER_${MODULE_NAME_UPPER}_MODULE_LOGIC_EXPORT
 
 set(${KIT}_INCLUDE_DIRECTORIES
   ${CMAKE_BINARY_DIR}/${MODULE_NAME}
+  ${CMAKE_CURRENT_BINARY_DIR}/..
```
Comment (RafaelPalomar) [inline VesselSegmentation/Logic/CMakeLists.txt]: Could we avoid the use of relative paths?

File: VesselSegmentation/Logic/itkVesselSegmentationPreProcessingFilter.h [hunk 2]
```diff
@@ -51,6 +57,13 @@ namespace itk
     class VesselSegmentationPreProcessingFilter:public ImageToImageFilter<  TInputImage, TOutputImage >
     {
     public:
+
+        #ifdef ITK_USE_GPU
+         bool itkGPU = true;
+        #else
+         bool itkGPU = false;
+        #endif
+
```
Comment (RafaelPalomar) [inline VesselSegmentation/Logic/itkVesselSegmentationPreProcessingFilter.h]: This is C++11 which should not be used in Slicer Extensions. As an alternative just declare the variable (why public??) and initialize the value in the constructor

File: VesselSegmentation/Logic/itkVesselSegmentationPreProcessingFilter.hxx [hunk 3]
```diff
@@ -76,15 +76,17 @@ namespace itk
...
         
         double timeStep = min_Spacing/(powf(2, 4)) - 0.0001; //4 = Dimension + 1; timeStep < min_Spacing/(powf(2, 4))
-        
-        typename GPUSmoothingFilterType::Pointer gpuSmoothing = GPUSmoothingFilterType::New();
-        gpuSmoothing->SetInput( remappedImage );
-        gpuSmoothing->SetTimeStep( timeStep );
-        gpuSmoothing->SetNumberOfIterations(  m_NumberOfIterations );
-        gpuSmoothing->SetConductanceParameter( m_Conductance );
+
+        std::cout << "USING GPU: " << itkGPU << std::endl;
```
Comment (RafaelPalomar) [inline VesselSegmentation/Logic/itkVesselSegmentationPreProcessingFilter.hxx]: This line is not useful. Remove it. If you want information about GPU usage do it as a message in the CMakeLists.txt

**Review by RafaelPalomar — CHANGES_REQUESTED** [review]
It looks good. I made a few comments (see below)

File: VesselSegmentation/Logic/CMakeLists.txt (hunk 1 above)
Comment (lcoram) [reply VesselSegmentation/Logic/CMakeLists.txt]: Don't know why this was even there, possibly left over from some problems integrating the filter initially. It has been deleted.

File: VesselSegmentation/Logic/itkVesselSegmentationPreProcessingFilter.h (hunk 2 above)
Comment (lcoram) [reply VesselSegmentation/Logic/itkVesselSegmentationPreProcessingFilter.h]: Cleaned this up, I should not have left it in... as I was just printing it out as a sanity check.

File: VesselSegmentation/Logic/itkVesselSegmentationPreProcessingFilter.hxx (hunk 3 above)
Comment (lcoram) [reply VesselSegmentation/Logic/itkVesselSegmentationPreProcessingFilter.hxx]: Deleted

**Comment by lcoram** [comment]
I have pushed another commit on this branch (think this updates the pull request?). I believe it addresses the comments you had.

**Pull request merged**